bridge/src/lib.rs

```diff
@@ -91,17 +92,21 @@ methods!(
         let request = unwrap_bytes(request.map_err(VM::raise_ex).unwrap());
         let metadata = to_hash_map(metadata.map_err(VM::raise_ex).unwrap());
         let timeout = timeout.map_or(None, |v| Some(v.to_u64()));
+        let (tx, rx) = mpsc::channel::<RpcResult>();
 
-        let result = Thread::call_without_gvl(move || {
+        let result = Thread::call_without_gvl(|| {
             let connection = _rtself.get_data_mut(&*CONNECTION_WRAPPER);
             let params = RpcParams {
                 rpc: rpc.clone(),
                 request: request.clone(),
                 metadata: metadata.clone(),
                 timeout_millis: timeout
             };
-            connection.call(params)
-        }, Some(|| {}));
+            connection.call(params, tx.clone());
+            rx.recv().expect("RPC result channel closed")
```

Comment:
I wouldn't use `expect` in either spot here. You could blow up the whole process if any specific call has some weird interaction where either closure is dropped before the other one finishes its channel operation. I'd return some error instead which just fails the call.

It's not immediately clear to me when the unblock function is called. If that happens during interrupt, then this mostly makes sense, but I think the fact that the channel can be sent to by either the call or the interrupt is a bit weird feeling. Namely, the actual spawned call won't be interrupted, it'll just lose the channel race.

Using cancel token like Chad suggested fixes that problem. That way you only ever send down the channel from inside the call, and you select over either the token getting canceled or the call resolving.

Reply:
Hah, this is all fixed while I was writing it!

Reply:
Yep, agree. I've also changed this to use cancellation token instead, but somehow you got the old version. Can you try refreshing the page?